src/DataTypes.d.ts

```diff
@@ -0,0 +1,35 @@
+import { firestore } from 'firebase-admin';
+
+type role = 'lead' | 'admin' | 'tpm' | 'pm' | 'developer' | 'designer';
+
+type Member = {
+  email: string;
+  first_name: string;
+  last_name: string;
+  role: string;
+  graduation: string;
+  major: string;
+  double_major?: string; //optional
+  minor?: string; //optional
+  website?: string; //optional
+  linkedin_link?: string; //optional
+  github_link?: string; //optional
+  hometown: string;
+  about: string;
+  subteam: string;
+  other_subteams?: string[]; // optional
+};
```

Comment:
so for the member type on the frontend right now, the type for the optional things like minor and double major is null | string (because input components don't work with undefined and firebase doesn't let you store a key with undefined as the value). Still not sure how we want to handle these optional values? What do you think?

Reply:
I don't think (?) there's anything wrong with just leaving them as null, so I guess it's probably fine that way